rand_chacha/src/chacha.rs

```diff
@@ -271,6 +283,78 @@ chacha_impl!(ChaCha20Core, ChaCha20Rng, 10, "ChaCha with 20 rounds");
 chacha_impl!(ChaCha12Core, ChaCha12Rng, 6, "ChaCha with 12 rounds");
 chacha_impl!(ChaCha8Core, ChaCha8Rng, 4, "ChaCha with 8 rounds");
 
+#[cfg(feature = "serde1")]
+impl Serialize for Array64<u32> {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: Serializer,
+    {
+        let mut state = serializer.serialize_struct("Array64<u32>", 1)?;
+        // must transmute to 32 length because serde can't serialize 64 length arrays yet
```

Comment:
Doesn't this have alignment issues?

I think the manual implementation can be avoided by using the `serde-big-array` crate.

Reply:
Yes I think it will be better with `serde-big-array` I'll see what its like.